src/pages/EditRequestPage.js

```diff
@@ -35,27 +36,74 @@ const defaultProps = {
     report: {},
 };
 
-function EditRequestPage(props) {
-    const parentReportAction = ReportActionsUtils.getParentReportAction(props.report);
-    const moneyRequestReportAction = ReportUtils.getMoneyRequestAction(parentReportAction);
-    const transactionDescription = moneyRequestReportAction.comment;
-    const field = lodashGet(props, ['route', 'params', 'field'], '');
+function EditRequestPage({report, route}) {
+    const transactionID = lodashGet(ReportActionsUtils.getParentReportAction(report), 'originalMessage.IOUTransactionID', '');
+    const transaction = TransactionUtils.getTransaction(transactionID);
+    const transactionDescription = TransactionUtils.getDescription(transaction);
+    const transactionAmount = TransactionUtils.getAmount(transaction);
+    const transactionCurrency = TransactionUtils.getCurrency(transaction);
 
-    function updateTransactionWithChanges(changes) {
-        // Update the transaction...
-        // eslint-disable-next-line no-console
-        console.log({changes});
+    // Take only the YYYY-MM-DD value
+    const transactionCreatedDate = new Date(TransactionUtils.getCreated(transaction));
+    const transactionCreated = format(transactionCreatedDate, CONST.DATE.FNS_FORMAT_STRING);
+    const fieldToEdit = lodashGet(route, ['params', 'field'], '');
 
-        // Note: The "modal" we are dismissing is the MoneyRequestAmountPage
+    // Update the transaction object and close the modal
+    function editMoneyRequest(transactionChanges) {
+        IOU.editMoneyRequest(transactionID, report.reportID, transactionChanges);
         Navigation.dismissModal();
     }
 
-    if (field === CONST.EDIT_REQUEST_FIELD.DESCRIPTION) {
+    if (fieldToEdit === CONST.EDIT_REQUEST_FIELD.DESCRIPTION) {
         return (
             <EditRequestDescriptionPage
                 defaultDescription={transactionDescription}
-                onSubmit={(changes) => {
-                    updateTransactionWithChanges(changes);
+                onSubmit={(transactionChanges) => {
+                    // In case the comment hasn't been changed, do not make the API request.
+                    if (transactionChanges.comment.trim() === transactionDescription) {
+                        Navigation.dismissModal();
+                        return;
+                    }
```

Comment:
NAB maybe we should add all these checks to `IOU.editMoneyRequest`

Reply:
We could add some verify method there

Reply:
I will add this to the clean ups